### ISSUE TITLE
Invariant Violation: Hooks can only be called inside the body of a function component

### ISSUE THREAD

**Issue by aralroca** [SylcatOfficial/react-form-uncontrolled#5]
Using `react-form-uncontrolled` library inside a React Component with react +`v.16.8.0-alpha-0` is working properly. However, testing the React Component that use `react-form-uncontrolled` inside, tests are failing for this reason:

>    Error: Uncaught [Invariant Violation: Hooks can only be called inside the body of a function component.] 

* **react**: 16.8.0-alpha-0
* **jest**: 23.6.0
* **react-testing-library**: 5.4.4

Looks that this issue only happens with the **transpiled** code (after Babel) in `build/index.js`. This is not happening using the code without the transpilation from `lib/index.js`...

In order to reproduce it, is not necessary to wrap the Form component of this library in another Component, we can just test it directly as:

```js
import React from 'react'
import { render } from 'react-testing-library'
import { Form } from 'react-form-uncontrolled'

describe('Form', () => {
  test('this test is going to break', () => {
    const { container } = render(<Form>test</Form>)

    expect(true).toBe(true)
  })
})
```

![image](https://user-images.githubusercontent.com/13313058/51079020-fbc3f980-16bf-11e9-9eb6-41d0f8ebcadd.png)

**Comment by aralroca** [comment]
I'm not sure if this problem is related to `react-form-uncontrolled` itself, or if it is more related to the `render` method of `react-testing-library.`

**Comment by aralroca** [comment]
I think the problem is not related to react-testing-library. Is related to this library.

Running a sandbox **without tests**, just using this library, also is getting the same error:

* https://codesandbox.io/s/znx2r55lkp

However, **WITHOUT TRANSPILATION** is working FINE (importing directly the code instead of using the library that transpile the code in the `build` folder... 
* https://codesandbox.io/s/jvr2pwv5pv

**Comment by aralroca** [comment]
Perhaps is nothing related with Babel transpilation... Looks that importing the transpiled code directly to the Sandbox is working without any problem:

* https://codesandbox.io/s/k0j6zn2xzo

**Comment by aralroca** [comment]
Related with: https://github.com/facebook/react/issues/14582

**Comment by aralroca** [comment]
Solved after fix the dependencies of the library `react-form-uncontrolled`. The problem was that **react** and **react-dom** was added as `dependencies` instead of `peerDependencies`. 

After fixing it, now is working well... No idea why happening exactly. Perhaps a problem of duplicated react libraries... 

* Fixed after **react-form-uncontrolled** `v0.3.2`: 
    * https://codesandbox.io/s/lrx20vw06m

Fixed by this commit: https://github.com/SylcatOfficial/react-form-uncontrolled/commit/4abc444c35e7464ea1f419c12bd4a025cb8902ae
